From patchwork Mon Nov 26 16:33:47 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Paolo Bonzini <pbonzini@redhat.com>
X-Patchwork-Id: 10698607
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 596B813BB
	for <patchwork-kvm@patchwork.kernel.org>;
 Mon, 26 Nov 2018 16:33:55 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 48FCF29051
	for <patchwork-kvm@patchwork.kernel.org>;
 Mon, 26 Nov 2018 16:33:55 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 3A43D29241; Mon, 26 Nov 2018 16:33:55 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.7 required=2.0 tests=BAYES_00,DKIM_INVALID,
	DKIM_SIGNED,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id C022829051
	for <patchwork-kvm@patchwork.kernel.org>;
 Mon, 26 Nov 2018 16:33:54 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1726869AbeK0D2a (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Mon, 26 Nov 2018 22:28:30 -0500
Received: from mail-wr1-f65.google.com ([209.85.221.65]:44895 "EHLO
        mail-wr1-f65.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1726343AbeK0D23 (ORCPT <rfc822;kvm@vger.kernel.org>);
        Mon, 26 Nov 2018 22:28:29 -0500
Received: by mail-wr1-f65.google.com with SMTP id z5so15311840wrt.11
        for <kvm@vger.kernel.org>; Mon, 26 Nov 2018 08:33:52 -0800 (PST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20161025;
        h=sender:from:to:subject:date:message-id;
        bh=iftFYIlaGpK3uWippCSd20oE9evfyxExkRtpJ0bUAHc=;
        b=YndJxfG6ekwyQcdag/iT/vP1NvIR1KkxG+li1/IZiPQeAbq8LDJ30xEcsgLyuxsJO+
         wTyHW9W0zCc9VRq8CPmGvkT9NljKln5CA6gDIP30jLIzjhRNYGRbkO6ZsfodKpdpjAbq
         sL63AKW+XeG93fN9yiIl2+PvcY4e01ZQdoHhnRPzrWZ3zDWjPq/KUVrWdnK17oMTNXY2
         c+gkWR4HcB3/4zktt/mRzVuMQNnWFrAy4Ly1nKRD6PmChftltbhZ6IN2rhGWdsoNggYd
         kz1v0MHCbdN32DS3k2nmhrhaeKLH0YCiSCUNHABNq15ckXBtaz6GKMEGXbEQTRA6zr2T
         wERA==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:sender:from:to:subject:date:message-id;
        bh=iftFYIlaGpK3uWippCSd20oE9evfyxExkRtpJ0bUAHc=;
        b=SxGoy+1DDltJlSShcZiFdHXqgZU80Fo6kkhsfWYfjgdnZKtjWezd4b72AQZtFE5pvC
         AwNM2KWqB0OEragblMXUbrOwvokAtkUQD9DUwP2leMn6ghdUQEc1V5dVWzUCsI/j1m0P
         zohJ+WMAAkALrPwPrm/KZxa6qGMfq5N6te9ykJrihQ8zFdDKxGXj1cyjJGNSf2WL6Oso
         1asU7ZGPJkAGLTJ1oBDfhvlcWIXWYkUIvajCL1dmsuD/0Mkf7utZ8bav10+O/YeCArOX
         ZvnVrdTIi6BL2LT1PdQvNlQzBYhaNBpv8gxVQEv8DdHAP9Pb1IDvrsEZ0URaVBNw0yTV
         +qlg==
X-Gm-Message-State: AA+aEWZWQi5I0ux1rPb2rESXguajnIuH/GkoG4KY6UOJXh8YjClZ9iB2
        50kaBQmU1JCkGSuwAkWCQFAQw1Cd
X-Google-Smtp-Source: 
 AFSGD/V0XnXvD64MCyI05bkSxbEJIDfYu9212yCxr/1kplqMbxdXPY5d7PcbbKlwTHKlcU56vg3NsQ==
X-Received: by 2002:adf:e3d0:: with SMTP id
 k16mr25671176wrm.223.1543250031167;
        Mon, 26 Nov 2018 08:33:51 -0800 (PST)
Received: from 640k.lan ([93.56.166.5])
        by smtp.gmail.com with ESMTPSA id
 198sm1819845wmj.38.2018.11.26.08.33.48
        for <kvm@vger.kernel.org>
        (version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
        Mon, 26 Nov 2018 08:33:48 -0800 (PST)
From: Paolo Bonzini <pbonzini@redhat.com>
To: kvm@vger.kernel.org
Subject: [PATCH kvm-unit-tests] vmx: fix infinite loop on vmentry failure
Date: Mon, 26 Nov 2018 17:33:47 +0100
Message-Id: <1543250027-16130-1-git-send-email-pbonzini@redhat.com>
X-Mailer: git-send-email 1.8.3.1
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

Broken by commit c76ddf062ecfc0b93c92f4006213888c487be92d
due to a copy-and-paste bug.

Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
---
 x86/vmx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/x86/vmx.c b/x86/vmx.c
index 9422b43..ab3c2a4 100644
--- a/x86/vmx.c
+++ b/x86/vmx.c
@@ -1605,7 +1605,7 @@ static bool vmx_enter_guest(struct vmentry_failure *failure)
 		"pushf\n\t"
 		"pop %%rdi\n\t"
 		"mov %%rdi, %[failure_flags]\n\t"
-		"movl $1, %[failure_flags]\n\t"
+		"movl $1, %[failure_early]\n\t"
 		"jmp 3f\n\t"
 		"vmx_return:\n\t"
 		SAVE_GPR_C
